quic/s2n-quic-transport/src/buffer/receive_buffer.rs

```diff
@@ -657,3 +657,16 @@ impl StreamReceiveBuffer {
         *self = StreamReceiveBuffer::with_buffer_size(self.buffer_size)
     }
 }
+
+#[test]
+#[cfg_attr(kani, kani::proof)]
+fn try_merge_receive_buffers() {
```

Comment:
Can you implement bolero harnesses for these instead? These harnesses have to be gated with `#[cfg(kani)]`, as it's not a `dev-dependency`. And my complaint about `#[cfg(kani)]` is:
* we don't get any type checking or analysis with a regular `cargo build`
* we won't find out this is broken until `cargo kani` is executed, which could be trying to debug in CI